src/products/views/ProductList/ProductList.tsx

```diff
@@ -300,21 +300,28 @@ export const ProductList: React.FC<ProductListProps> = ({ params }) => {
     }),
     [params, settings.rowNumber]
   );
-  const { data, loading, refetch } = useProductListQuery({
-    displayLoader: true,
-    variables: { ...queryVariables, hasChannel: !!selectedChannel }
-  });
 
   function filterColumnIds(columns: ProductListColumns[]) {
     return columns
       .filter(isAttributeColumnValue)
       .map(getAttributeIdFromColumnValue);
   }
+  const filteredColumnIds = filterColumnIds(settings.columns);
+
+  const { data, loading, refetch } = useProductListQuery({
+    displayLoader: true,
+    variables: {
+      ...queryVariables,
+      hasChannel: !!selectedChannel,
+      hasSelectedAttributes: !!filteredColumnIds.length
```

Comment:
```suggestion
      hasSelectedAttributes: filteredColumnIds.length > 0
```
makes it slightly more readable when skimming through code